src/main.rs

```diff
@@ -238,13 +325,73 @@ fn main() -> NazaraResult<()> {
     // Welcome Message.
     println!("{ASCII_ART}");
 
-    // Collect machine information.
-    let machine = Machine {
-        name: None,
-        dmi_information: dmi::construct_dmi_information()?,
-        network_information: network::construct_network_information()?,
-        custom_information: Some(execute(args.plugin)?),
-    };
+    match &args.command {
+        Commands::WriteConfig {
+            uri,
+            token,
+            name,
+            description,
+            comments,
+            status,
+            device_type,
+            site,
+            role,
+            cluster_id,
+            json,
+        } => {
+            println!("Writing configuration file...");
+            if let Some(json_str) = json {
```

Comment:
If you follow the suggestion on R355, you can do this and achieve the same effect:

```suggestion
            if json.is_some() {
```